packages/openapi-generator/src/options/per-service-config.ts

```diff
@@ -0,0 +1,66 @@
+import { existsSync, promises } from 'fs';
+import { relative } from 'path';
+
+const { readFile } = promises;
+
+/**
+ * Represents the service configuration for all services.
+ */
+export interface PerServiceConfig {
+  /**
+   * Configuration for a service, specified by its filename.
+   */
+  [fileName: string]: ServiceConfig;
+}
+
+/**
+ * Represents the service configuration for one service.
+ * This is usually specified as part of the per service configuration.
+ */
+export interface ServiceConfig {
+  /**
+   * Name of the directory to store the service in.
+   */
+  directoryName: string;
+  /**
+   * Name of the package to reference in the package.json.
+   */
+  packageName: string;
+}
+
+/**
+ * Get the per service configuration from the given path.
+ * @param configPath Path to the given per service configuration or undefined if no path was given.
+ * @returns The parsed configuration for all services.
+ */
+export async function getPerServiceConfig(
+  configPath: string | undefined
+): Promise<PerServiceConfig | undefined> {
+  if (configPath) {
+    return existsSync(configPath)
+      ? JSON.parse(await readFile(configPath, 'utf8'))
+      : {};
+  }
+}
+
+export function getOrCreateServiceConfig(
+  perServiceConfig: PerServiceConfig | undefined,
+  inputFilePath: string,
+  serviceName: string
+): ServiceConfig {
+  const relativeFilePath = relative(process.cwd(), inputFilePath);
```

Comment:
Should we consider doing a spread here i.e. adding all non existing keys?

Reply:
https://github.com/SAP/cloud-sdk-js/pull/1249#discussion_r622987222

Reply:
Will consider this or create a follow up.

Reply:
check